client/src/App.jsx

```diff
@@ -1,34 +1,39 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { useState, useEffect } from 'react'
 import './App.css'
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [responseData, setResponseData] = useState(null);
+
+  function handleSubmit(e) {
+    e.preventDefault();
+
+    const form = e.target;
+    const formData = new FormData(form);
+
+    fetch('http://localhost:5000/api/threads', { method: "post", body: formData })
```

Comment:
You are correct. 

Just a suggestion, more "modern" way to do the same in JavaScript are the followings:

1. Declare you function as a `async` function
2. After that, you can use the `async`/`await` syntax to play with the `Promise`.
```
try {
    const response = await fetch('http://localhost:5000/api/threads', { method: "post", body: formData });
    const data = await response.json()
    console.log(data);
    setResponseData(data);
} catch (error) {
    console.log('There was an error', error);
}
```